### PR TITLE
Packages Page

### PR DESCRIPTION
**Pull request by jhuggett**
Adding package pages to the docs by pulling the READMEs from GitHub.

### CODE REVIEW

File: pages/packages/index.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+          previewError: { ...e }, //workaround since we cant return error as JSON
+        },
+      }
+    } else {
+      throw e
+    }
+  }
+}
+
+export default DocTemplate
```
Comment (ncphillips) [inline pages/packages/index.tsx:22]: That's super interesting

Comment (jhuggett) [reply pages/packages/index.tsx:22]: In a good way or a bad way?

File: utils/docs/getPackageProps.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2)
+        })
+        .join('')
+    )
+}
+
+export async function getPackageProps({ preview, previewData }: any, slug: string) {
+
+    const filePath = path.join(process.cwd(), 'content/packages.json')
+    const file = JSON.parse(fs.readFileSync(filePath, 'utf8'))
```
Comment (ncphillips) [inline utils/docs/getPackageProps.ts:25]:
```suggestion
    const file = await readJsonFile('content/packages.json')
```

File: pages/packages/index.tsx (hunk 1 above)
Comment (ncphillips) [reply pages/packages/index.tsx:22]: Good, I just didn't even think about re-using a template like that

File: utils/docs/getPackageProps.ts (hunk 2 above)
Comment (jhuggett) [reply utils/docs/getPackageProps.ts]: It does, just had to export it

File: pages/packages/[slug].tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import fs from 'fs'
+import { getPackageProps } from '../../utils/docs/getPackageProps'
+import { DocsNavToggle, DocsMobileTinaIcon, DocsContent } from 'pages/docs/[...slug]'
+
+  
+  export default function Packages(props) {
+    const frontmatter = {
+      title: 'packages',
+    }
+    const excerpt = 'A package for Tinacms.'
```
Comment (DirtyF) [inline pages/packages/[slug].tsx:27]: TinaCMS

**Comment by kendallstrautman** [comment]
The layout and styles need to match the docs template. And I think we should add the title so it's consistent with the other docs...so we'll need to add the fm value in the READMEs. It would also be awesome to add the TOC if possible.

**Comment by spbyrne** [comment]
I can handle the layout changes & adding the ToC. There's a lot of code to duplicate (I copied the new layout to guides) so I may refactor to remove that duplication for pages using docs layout/styling.

**Comment by kendallstrautman** [comment]
Beat ya to the layout stuff - https://github.com/tinacms/tinacms.org/pull/570 lol but yeah let's do the TOC on another PR. I think you're right, there's some stuff we could extract 👌

I think all that's left is to have the title source from the frontmatter of the README file and remove those h1s on the READMEs

**Comment by kendallstrautman** [comment]
Also @jhuggett do you have prettier set up? Just seeing some formatting weirdness

**Comment by jhuggett** [comment]
@kendallstrautman I have prettier installed but I probably don't have it configured correctly

**Comment by kendallstrautman** [comment]
lol I feel ya..... @spbyrne & @ncphillips can attest to my pains with prettier 🤦‍♀️

**Comment by ncphillips** [comment]
This looks good to me. We can add more packages in separate PRs. Are we good to go here or are we waiting on a couple things?

**Comment by kendallstrautman** [comment]
@ncphillips the only thing is sourcing the frontmatter title from the .md file, but we could also do that in another PR

**Comment by kendallstrautman** [comment]
Oh and we should merge my layout adjustments

**Pull request merged**